bug_update.php

```diff
@@ -243,7 +250,7 @@
 }
 
 # Ensure that the user has permission to change the view status of the issue.
-if( $t_existing_bug->view_state !== $t_updated_bug->view_state ) {
+if( (int)$t_existing_bug->view_state !== $t_updated_bug->view_state ) {
```

Comment:
Seems this introduced a regression
https://www.mantisbt.org/bugs/view.php?id=20478